src/opm/parser/eclipse/EclipseState/Schedule/UDQ/UDQFunction.cpp

```diff
@@ -0,0 +1,509 @@
+        auto& udq_value = result[index];
+        if (udq_value)
+            result.assign( index, std::fabs(udq_value.value()));
+    }
+    return result;
+}
+
+UDQSet UDQUnaryElementalFunction::DEF(const UDQSet& arg) {
+    auto result = arg;
+    for (std::size_t index=0; index < result.size(); index++) {
+        auto& udq_value = result[index];
+        if (udq_value)
+            result.assign( index, 1 );
+    }
+    return result;
+}
+
+
+UDQSet UDQUnaryElementalFunction::UNDEF(const UDQSet& arg) {
+    UDQSet result(arg.size());
+    for (std::size_t index=0; index < result.size(); index++) {
+        auto& udq_value = arg[index];
+        if (!udq_value)
+            result.assign( index, 1 );
+    }
+    return result;
+}
+
+
+UDQSet UDQUnaryElementalFunction::IDV(const UDQSet& arg) {
+    auto result = arg;
+    for (std::size_t index=0; index < result.size(); index++) {
+        auto& udq_value = result[index];
+        if (udq_value)
+            result.assign( index, 1 );
+        else
+            result.assign(index , 0);
+    }
+    return result;
+}
+
+UDQSet UDQUnaryElementalFunction::EXP(const UDQSet& arg) {
+    auto result = arg;
+    for (std::size_t index=0; index < result.size(); index++) {
+        auto& udq_value = result[index];
+        if (udq_value)
+            result.assign( index, std::exp(udq_value.value()) );
+    }
+    return result;
+}
+
+UDQSet UDQUnaryElementalFunction::NINT(const UDQSet& arg) {
+    auto result = arg;
+    for (std::size_t index=0; index < result.size(); index++) {
+        auto& udq_value = result[index];
+        if (udq_value)
+            result.assign( index, std::nearbyint(udq_value.value()) );
+    }
+    return result;
+}
+
+
+UDQSet UDQUnaryElementalFunction::RANDN(std::mt19937& rng, const UDQSet& arg) {
+    auto result = arg;
+    std::normal_distribution<double> dist(0,1);
+    for (std::size_t index=0; index < result.size(); index++) {
+        auto& udq_value = result[index];
+        if (udq_value)
+            result.assign( index, dist(rng) );
+    }
+    return result;
+}
+
+
+UDQSet UDQUnaryElementalFunction::RANDU(std::mt19937& rng, const UDQSet& arg) {
+    auto result = arg;
+    std::uniform_real_distribution<double> dist(-1,1);
+    for (std::size_t index=0; index < result.size(); index++) {
+        auto& udq_value = result[index];
+        if (udq_value)
+            result.assign( index, dist(rng) );
+    }
+    return result;
+}
+
+
+
+
+
+UDQSet UDQUnaryElementalFunction::LN(const UDQSet& arg) {
+    auto result = arg;
+    for (std::size_t index=0; index < result.size(); index++) {
+        auto& udq_value = result[index];
+        if (udq_value) {
+            double arg = udq_value.value();
+            if (arg > 0)
+                result.assign(index, std::log(arg));
+            else
+                throw std::invalid_argument("Argument must be > 0 for LN function invalid: " + std::to_string(arg));
```

Comment:
I don't understand that message.  I believe there is punctuation and/or words missing.